src/main/java/org/opensearch/neuralsearch/processor/ScoreNormalizer.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Optional;
+
+import lombok.extern.log4j.Log4j2;
+
+import org.apache.lucene.search.ScoreDoc;
+import org.apache.lucene.search.TopDocs;
+import org.opensearch.neuralsearch.search.CompoundTopDocs;
+
+/**
+ * Abstracts normalization of scores in query search results.
+ */
+@Log4j2
+public class ScoreNormalizer {
```

Comment:
I am not super convinced in having this class. I think we should move the actual min,max score calculation and updating the scores to ScoreNormalizationTechnique enum class. Over there too, I would like to split the code in smaller functions like getMinScore, get MaxScore, update Score for TopDocs so that they can be used across Normalization Technique.


The current code is very specific to Min/Max normalization and doesn't seems like it can be extended easily.